pynicotine/gtkgui/widgets/dialogs.py

```diff
@@ -230,3 +230,89 @@ def option_dialog(parent, title, message, callback, callback_data=None,
         self.add_button(third, Gtk.ResponseType.REJECT)
 
     self.present_with_time(Gdk.CURRENT_TIME)
+
+
+def custom_dialog(parent, title, message, callback, callback_data=None, cancel=True, selectable=True, third="",
+                  id_0="", id_1="", id_2="", id_3="", id_4="", id_5="", id_6="", id_9="", default_button=1,
+                  tip_0="", tip_1="", tip_2="", tip_3="", tip_4="", tip_5="", tip_6="", tip_9="",
+                  checkbox_label="", checkbox_tip="", checkbox_value=False):
+
+    """ Custom Dialog (direct replacement for option_dialog and message_dialog. ToDo: entry_dialog integration) """
+
+    if id_0 or id_1:
+        # Use Custom Buttons (0, 1... 6, 9)
+        buttons = Gtk.ButtonsType.NONE
+        icon = Gtk.MessageType.QUESTION
+
+    elif cancel and (id_0 == id_1 == ""):
+        # Standard option_dialog
+        buttons = Gtk.ButtonsType.OK_CANCEL
+        icon = Gtk.MessageType.QUESTION
+    else:
+        # Simple message_dialog
+        buttons = Gtk.ButtonsType.OK
+        icon = Gtk.MessageType.INFO
+
+    self = Gtk.MessageDialog(
+        transient_for=parent,
+        message_type=icon,  # INFO; WARNING; QUESTION; ERROR; OTHER
+        buttons=buttons,
+        text=title,
+        secondary_text=message or ""
+    )
+    self.connect("response", callback, callback_data)
+    self.set_destroy_with_parent(True)
+    self.set_modal(True)
+
+    if Gtk.get_major_version() == 4:
+        label = self.get_message_area().get_last_child()
+    else:
+        label = self.get_message_area().get_children()[-1]
+
+    label.set_selectable(selectable)
+
+    # Checkbox Option
+    if checkbox_label:
+        self.checkbox = Gtk.CheckButton()
+        self.checkbox.set_label(checkbox_label)
+        self.checkbox.set_active(checkbox_value)
+        self.checkbox.set_tooltip_text(checkbox_tip)
+
+        self.get_message_area().add(self.checkbox)
+        self.checkbox.show()
+    else:
+        self.checkbox = None  # option can be hidden
```

Comment:
> What else is missing from the existing dialogs?

The ability to be able to hide the checkbox, because NoneType isn't possible in the callback response.